ClientApp/src/components/AddListing.js

```diff
@@ -43,67 +67,95 @@ export default function AddListing(props) {
+                <div className="form-group">
+                    <label>Quantity</label>
+                    <input type="number" className="form-control" {...register("Quantity", { required: "Quantity of minimum 1 is required.", min: 1, max: 100 })} />
+                </div>
+                <div className="form-group">
+                    <label>Country</label>
+                    <input className="form-control" {...register("Location.Country", { required: true, message: "Country is required." })} />
+                    <label>District</label>
+                    <input className="form-control" {...register("Location.District")} />
+                    <label>City</label>
+                    <input className="form-control" {...register("Location.City", { required: true, message: "City is required." })} />
+                    <label>Radius in kilometers</label>
+                    <input type="number" className="form-control" {...register("Location.RadiusKM", { required: true, message: "Radius of minimum 1 km is required.", min: 1, max: 100 })} />
+                    <small className="form-text text-muted"> Radius describes how close you are located from the city center.</small>
+                </div>
+                <div className="form-group">
+                    <label>Images</label><br />
+                    <input asp-for="FileUpload.FormFile" type="file" className="form-control-file" name="temp-image" accept="image/*"{...register("ImagePath")} />
+                </div><br />
+                <button disabled={isSubmitting} className="btn btn-primary mr-1">
+                    {isSubmitting && <span className="spinner-border spinner-border-sm mr-1"></span>}
+                    Submit
+                </button>
+                {/* {isSubmitted && <Redirect push to='/' />} */}
+            </form>
+            <Modal isOpen={plantModal} toggle={togglePlantModal} >
+                <ModalHeader><Label>Choose a plant to add to your garden</Label></ModalHeader>
+                <ModalBody>
+                    <Row>
+                        {Object.entries(Plants).map(([key, p], ind) => <Col key={ind}>
+                            <Button onClick={() => {
+                                setData(Object.assign({ gardenPlant: key }, Data));
+                                setPlantModal(false);
+                                setPlaceModal(true);
+                            }} style={{ display: 'flex', justifyContent: 'center', margin: 'auto' }}><DisplayPlant Plant={p} /></Button>
+                        </Col>)}
+                    </Row>
+                </ModalBody>
+            </Modal>
+            <Modal isOpen={placeModal} toggle={togglePlaceModal} >
+                <ModalHeader><Label>Choose a place to situate your plant</Label></ModalHeader>
+                <ModalBody>
+                    <GardenComp username={user.username} onPlaceChosen={onPlaceChosen} />
+                </ModalBody>
+            </Modal>
```

Comment:
Could add a simple back button like an 

**X**

 to the Modal window so the window could be closed in case you want to still edit your listing before posting it.